mlflow/transformers/__init__.py

```diff
@@ -1326,10 +1336,7 @@ def _load_component(root_path: pathlib.Path, component_key: str, component_type)
     return component_instance.from_pretrained(component_path)
 
 
-def _generate_base_flavor_configuration(
-    pipeline,
-    task: str,
-) -> Dict[str, str]:
+def _generate_base_flavor_configuration(pipeline: Pipeline) -> Dict[str, str]:
```

Comment:
Agreed. While the previous behavior was more 'flexible' from the user perspective, I think it might open the door for potentially unexpected inference results. Let's go with the simpler and less risky approach here.